### PR TITLE
prepare backend for #204 

### PR DESCRIPTION
**Pull request by rnwgnr**
I've  added the possibility to download small and medium size images as requested in #204 

Pull request for lychee-front is upcoming, need to do some testing first.

### CODE REVIEW

File: php/Modules/Photo.php [hunk 1]
```diff
@@ -1204,25 +1205,37 @@ public function getArchive() {
...
+			default:
+				$filepath = LYCHEE_UPLOADS_BIG;
+		}
+		
+		$fullfilepath = $filepath . $photo->url;
 		// Check the file actually exists
-		if (!file_exists(LYCHEE_UPLOADS_BIG . $photo->url)) {
-			Log::error(Database::get(), __METHOD__, __LINE__, 'Original photo missing: ' .LYCHEE_UPLOADS_BIG .$photo->url);
+		if (!file_exists($fullfilepath)) {
+			Log::error(Database::get(), __METHOD__, __LINE__, 'Original photo missing: ' .$fullfilepath);
```
Comment (ildyria) [inline php/Modules/Photo.php]: This part of the warning should be changed: "Original photo"

**Review by ildyria — CHANGES_REQUESTED** [review]
I pretty okay with all this changes. I would just like to highlight the fact that the download function has been removed from the v4 (as the user can just right click to download the picture.

Warning needs to be fixed.

**Comment by d7415** [comment]
> I would just like to highlight the fact that the download function has been removed from the v4 (as the user can just right click to download the picture.

On that basis, I have to wonder whether it would be simpler to go without the download function on either (at least for small/medium) as that would remove any need to change Lychee v3 - everything could be done in the front end and (presumably - I'm still not entirely comfortable with v4) be more consistent.

**Comment by rnwgnr** [comment]
I'm currently evaluating switching from Piwigo to Lychee, an easy download function for users to get full, medium or small sized photos would be vital to me.
How is a user supposed to get the links to medium and small sized versions of the picture when removing the download feature?

**Review by ildyria — APPROVED** [review]

**Comment by mir07** [comment]
Could this be extended to also make download of the raw file available for download provided a method to upload raw files is available?

**Comment by rnwgnr** [comment]
@mir07 
This could be extended easily to support other formats as well (e.g. the thumbnail). The bigger part would be to allow raw file handling with Lychee. You should open a new issue for a discussion about this.

@ildyria @d7415 
I hope to send the pull request for the frontend tomorrow or on wednesday at the latest.

**Comment by ildyria** [comment]
@guzzisti This is currently not implemented in Lychee v4 ( https://github.com/LycheeOrg/Lychee-Laravel/issues/95 ). If you want to do it, it would be a nice step for you to actually learn how the new architecture works. :)

**Review by ildyria — APPROVED** [review]
Looks good to me.

**Comment by rnwgnr** [comment]
@ildyria will have a look at the laravel implementation later.

**Pull request merged**